packages/compiler/server/serverlib.ts

```diff
@@ -518,6 +524,28 @@ export function createServer(host: ServerHost): Server {
     return symbols;
   }
 
+  async function findDocumentHighlight(
+    params: DocumentHighlightParams
+  ): Promise<DocumentHighlight[]> {
+    const file = await compilerHost.readFile(await getPath(params.textDocument));
+    const identifiers = await compile(params.textDocument, (program, document, file) =>
+      findReferenceIdentifiers(program, file, document.offsetAt(params.position), false)
+    );
+    const highlight: DocumentHighlight[] = [];
+    if (identifiers === undefined) {
+      return highlight;
+    }
+    for (let i = 0; i < identifiers.length; i++) {
```

Comment:
here you should be able to replace with `return identifiers.map(identifier => ` instead of a for loop with index

Reply:
sounds good